Doc/library/os.rst

```diff
@@ -569,6 +569,38 @@ process and user.
       See the documentation for :func:`getgroups` for cases where it may not
       return the same group list set by calling setgroups().
 
+.. function:: setns(fd, nstype=0)
+
+   Reassociate the current with a namespace.
+   See the :manpage:`setns(2)` man page for more details.
+
+   If *fd* refers to a ``/proc/[pid]/ns/`` link, ``setns()`` reassociates the
+   calling thread with the namespace associated with that link, subject to any
+   constraints imposed by the *nstype* argument (a *nstype* of ``0`` means no
+   constraints).
+   Since Linux 5.8, *fd* may refer to a PID file descriptor obtained from
+   :func:`~os.pidfd_open`. In this case ``setns()`` reassociates the calling thread
+   into one or more of the same namespaces as the thread referred to by *fd*
+   subject to any constraints imposed by *nstype*, which is a bit mask specified
+   by combining one or more of the ``CLONE_NEW*`` constants,
+   e.g., as ``setns(fd, os.CLONE_NEWUTS | os.CLONE_NEWPID)``.
```

Comment:
I personally found this passage rather confusing and hard to follow, especially given the length and density of the sentence and, in particular, the information about ``nstype`` being split up between two different cases (fd is a link vs PID file descriptor)—I was very unsure about whether the information applied to both, or only the case in question.

Assuming that it applies to both, I've reorganized this a bit to keep the sentences shorter and more focused, and avoid duplicating and splitting up the ``nstype`` information, as well as some smaller fixes and clarifications.

```suggestion
   calling thread with the namespace associated with that link. 
   Since Linux 5.8, *fd* may refer to a PID file descriptor obtained from
   :func:`~os.pidfd_open`.
   In that case, ``setns()`` reassociates the calling thread
   into one or more of the same namespaces as the thread referred to by *fd*.
   In both cases, this is subject to any constraints imposed by the *nstype*
   argument, which is a bit mask specified by
   combining one or more of the ``CLONE_NEW*`` constants,
   e.g. ``setns(fd, os.CLONE_NEWUTS | os.CLONE_NEWPID)``.
   A *nstype* of ``0``, the default, means no constraints.
```

If this is not the case, I can propose an alternate revision given clear information on which aspects of `nstype` apply to which.

Reply:
`nstype` is not a bit mask in both cases. In the the first case (a link to a `/proc/pid/ns`), nstype can have one of the constants as value, in the second case (PID fd) it can be a bitmask of many namespaces to enter

Reply:
I see; it is even more unclear to me that such was the intent, then, and the valid values in the first case are never explicitly stated at all. As mentioned above, I will propose a revision to clarify it accordingly along with incorporating the other fixes above.